BeeSwift/Settings/ConfigureNotificationsViewController.swift

```diff
@@ -78,47 +84,113 @@ class ConfigureNotificationsViewController: UIViewController {
     }
     
     @objc func foregroundEntered() {
+        self.fetchGoals()
         self.updateHiddenElements()
     }
     
-    func fetchGoals() {
+    @objc func fetchGoals() {
         Task { @MainActor in
-            let goals = ServiceLocator.goalManager.staleGoals(context: ServiceLocator.persistentContainer.viewContext) ?? []
-            self.goals = goals.sorted { (goal1, goal2) -> Bool in
-                return goal1.slug > goal2.slug
+            self.tableView.refreshControl?.endRefreshing()
+
+            MBProgressHUD.showAdded(to: self.view, animated: true)
+            do {
+                try await ServiceLocator.goalManager.refreshGoals()
+                self.goals = ServiceLocator.goalManager.staleGoals(context: ServiceLocator.persistentContainer.viewContext)?.sorted(by: { $0.slug < $1.slug }) ?? []
+                self.lastFetched = Date()
+                MBProgressHUD.hide(for: self.view, animated: true)
+            } catch {
+                logger.error("Failure fetching goals: \(error)")
+
+                MBProgressHUD.hide(for: self.view, animated: true)
+                if UIApplication.shared.applicationState == .active {
+                    let alert = UIAlertController(title: "Error fetching goals", message: error.localizedDescription, preferredStyle: .alert)
+                    alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+                    self.present(alert, animated: true, completion: nil)
+                }
             }
             self.tableView.reloadData()
         }
     }
 }
 
+private extension ConfigureNotificationsViewController {
+    var goalsUsingDefaultNotifications: [Goal] {
+        self.goals.filter { $0.useDefaults }
+    }
+    
+    var goalsUsingNonDefaultNotifications: [Goal] {
+        self.goals.filter { !$0.useDefaults }
+    }
+
+}
+
 extension ConfigureNotificationsViewController : UITableViewDataSource, UITableViewDelegate {
     func numberOfSections(in tableView: UITableView) -> Int {
-        return 2
+        guard lastFetched != nil else { return 0 }
+        return 3
     }
     
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        if section == 1 { return self.goals.count }
-        return 1
+        switch section {
+        case 0:
+            return 1
+        case 1:
+            return self.goalsUsingDefaultNotifications.count
+        default:
+            return self.goalsUsingNonDefaultNotifications.count
```

Comment:
Non blocking, but I'm a fan of explicitly handling all the expected cases and have `default:` log/report an error.